### PR TITLE
NAS-120582 / 23.10 / Improve the `transfers` option

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

On **Data Protection > Add Cloud Sync Task**, test saving/loading of new **Transfers** selector

### CODE REVIEW

File: src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts [hunk 1]
```diff
@@ -407,6 +419,26 @@ export class CloudsyncFormComponent {
         }
       }
     });
+
+    this.form.controls.transfers.valueChanges.pipe(untilDestroyed(this)).subscribe((value: number) => {
+      const transfersDefaultValues = this.transfersDefaultOptions.map((transfers) => transfers.value);
```
Comment (undsoft) [inline src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts:424]: This method hard to read. Can we split it and maybe merge with code below for showing Custom option.

File: src/app/pages/data-protection/cloudsync/custom-transfers-dialog/custom-transfers-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+@UntilDestroy()
+@Component({
+  templateUrl: './custom-transfers-dialog.component.html',
+  styleUrls: ['./custom-transfers-dialog.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CustomTransfersDialogComponent {
+  readonly helptext = helptext;
+  readonly transfers = this.formBuilder.control([null as number]);
```
Comment (undsoft) [inline src/app/pages/data-protection/cloudsync/custom-transfers-dialog/custom-transfers-dialog.component.ts:15]: Needs required validation and min number validation.

File: src/app/pages/data-protection/cloudsync/custom-transfers-dialog/custom-transfers-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  <ix-form-actions>
+    <button mat-button type="button" matDialogClose ixTest="cancel">
+      {{ 'Cancel' | translate }}
+    </button>
+
+    <button
+      mat-button
+      color="primary"
+      type="button"
+      ixTest="restore"
```
Comment (undsoft) [inline src/app/pages/data-protection/cloudsync/custom-transfers-dialog/custom-transfers-dialog.component.html:21]: ixTest="save"

**Review by undsoft — APPROVED** [review]
`transfers` is being sent though I had troubles submitting form. This is likely unrelated to current PR and I will create another ticket.

**Comment by bvasilenko** [comment]
backport

**Pull request merged**